src/helpers.js

```diff
@@ -2,6 +2,27 @@
 import { spawn } from 'child_process'
 import * as core from "@actions/core"
 import fs from 'fs'
+import os from 'os'
+
+/**
+ * @returns {string}
+ */
+export const getOptionalSudoPrefix = () => {
+  switch(core.getInput("sudo")){
+    case "true":{
+      return "sudo ";
+    }
+    case "detect":{
+      return os.userInfo().uid == 0 ? "" : "sudo ";
+    }
+    case "false":{
+      return "";
+    }
+    default:{
+      throw new Error(`Unsupported 'sudo' value: ${core.getInput("sudo")}`);
+    }
+  }
```

Comment:
I would prefer this to be more succinct, like this:

```js
const useSudoPrefix = () => {
  const input = core.getInput("sudo")
  return input === "auto" ? external_os_default().userInfo().uid !== 0 : input === "true"
}
```

Reply:
This discards the error message, but I don't think he's that important, I'm going to do it